app/uk/gov/hmrc/timetopay/arrangement/services/PollerService.scala

```diff
@@ -94,14 +104,13 @@ class PollerService @Inject() (
       val desSubmissionRequest: DesSubmissionRequest = arrangement.desArrangement
         .getOrElse(throw new RuntimeException("Retry poller - Saved ttp in work item repo had no desArrangement to send " + wi.toString))
 
-      desArrangementApiServiceConnector.submitArrangement(utr, desSubmissionRequest).map {
+      desArrangementApiServiceConnector.submitArrangement(utr, desSubmissionRequest).flatMap {
```

Comment:
with `map`s here and below, the `process()` call will happen while the `tryDesCallAgain` returns a completed `Future` which means calls to DES could still be happening when we've reached the `onComplete` block of the current scheduled run.